src/error.rs

```diff
@@ -1,64 +1,81 @@
 use base64::DecodeError;
-use std::{num::ParseIntError, str};
+use reqwest::Error as ReqwestError;
+use serde_json::Error as JsonError;
+use std::io::Error as IoError;
+use std::num::ParseIntError;
+use std::str::Utf8Error;
 use thiserror::Error;
 use websocket::{client::ParseError, WebSocketError};
 
 /// Enumeration of all possible errors in the `socket.io` context.
 #[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum Error {
+    // Conform to https://rust-lang.github.io/api-guidelines/naming.html#names-use-a-consistent-word-order-c-word-order
+    // Negative verb-object
     #[error("Invalid packet id: {0}")]
     InvalidPacketId(u8),
     #[error("Error while parsing an empty packet")]
-    EmptyPacket,
+    EmptyPacket(),
     #[error("Error while parsing an incomplete packet")]
-    IncompletePacket,
+    IncompletePacket(),
     #[error("Got an invalid packet which did not follow the protocol format")]
-    InvalidPacket,
+    InvalidPacket(),
     #[error("An error occurred while decoding the utf-8 text: {0}")]
-    Utf8Error(#[from] str::Utf8Error),
+    InvalidUtf8(#[from] Utf8Error),
     #[error("An error occurred while encoding/decoding base64: {0}")]
-    Base64Error(#[from] DecodeError),
+    InvalidBase64(#[from] DecodeError),
     #[error("Invalid Url: {0}")]
     InvalidUrl(String),
     #[error("Error during connection via http: {0}")]
-    ReqwestError(#[from] reqwest::Error),
+    IncompleteReqwest(#[from] ReqwestError),
     #[error("Network request returned with status code: {0}")]
-    HttpError(u16),
+    IncompleteHttp(u16),
     #[error("Got illegal handshake response: {0}")]
-    HandshakeError(String),
+    InvalidHandshake(String),
     #[error("Called an action before the connection was established")]
-    ActionBeforeOpen,
+    IllegalActionBeforeOpen(),
     #[error("string is not json serializable: {0}")]
-    InvalidJson(#[from] serde_json::Error),
+    InvalidJson(#[from] JsonError),
     #[error("Did not receive an ack for id: {0}")]
-    DidNotReceiveProperAck(i32),
+    MissingAck(i32),
     #[error("An illegal action (such as setting a callback after being connected) was triggered")]
-    IllegalActionAfterOpen,
+    IllegalActionAfterOpen(),
     #[error("Specified namespace {0} is not valid")]
     IllegalNamespace(String),
     #[error("A lock was poisoned")]
-    PoisonedLockError,
+    InvalidPoisonedLock(),
     #[error("Got a websocket error: {0}")]
-    FromWebsocketError(#[from] WebSocketError),
+    IncompleteWebsocket(#[from] WebSocketError),
```

Comment:
Mhh same with the `reqwestError`. Maybe there is a way to indicate that this error originates from one of our dependecies.

Reply:
```suggestion
    IIncompleteResponceFromWebsocket(#[from] WebSocketError),
```

Reply:
Makes sense!